jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -143,13 +169,82 @@ public static Resource.Mount mount(URI uri) throws IOException
      */
     public static Resource.Mount mountJar(Path path) throws IOException
     {
+        if (!isJar(path))
+            throw new IllegalArgumentException("Path is not a JAR: " + path);
         URI pathUri = path.toUri();
         if (!pathUri.getScheme().equalsIgnoreCase("file"))
             throw new IllegalArgumentException("not an allowed path: " + path);
         URI jarUri = URI.create(toJarRoot(pathUri.toString()));
         return FileSystemPool.INSTANCE.mount(jarUri);
     }
 
+    public static boolean isJar(Path path)
+    {
+        if (path == null)
+            return false;
+        return path.getFileName().toString().toLowerCase(Locale.ENGLISH).endsWith(".jar");
+    }
+
+    public static boolean isJar(URI uri)
+    {
+        if (uri == null)
+            return false;
+        if (uri.getScheme() == null)
+            return false;
+        String path = uri.getPath();
+        if (uri.getScheme().equalsIgnoreCase("jar"))
+        {
+            String ssp = uri.getSchemeSpecificPart();
+            path = URI.create(ssp).getPath();
+        }
+        if (path == null)
+            return false;
+        // look for `!/` split
+        int idxEnd = path.indexOf("!/");
+        if (idxEnd < 0)
+            idxEnd = path.length();
+        // look for filename suffix
+        int idxSuffix = path.lastIndexOf('.', idxEnd);
+        if (idxSuffix < 0)
+            return false; // no suffix found, can't possibly be a jar/war
+        String suffix = path.substring(idxSuffix, idxEnd).toLowerCase(Locale.ENGLISH);
+        return suffix.equals(".jar") || suffix.equals(".war");
+    }
+
+    public static URI toJarRoot(URI uri)
```

Comment:
There is something wierd with the return of this method, which I  think it is something like "jar:file:/some/path/foo.jar", which is not valid as it is missing the "!/".

It should return either "file:/some/path/foo.jar" OR "jar:file:/some/path/foo.jar!/".
I don't mind either, but have a slight preference for the later as it matches the name better.

Eitherway, javadoc is needed.